### DISCUSSION TITLE
Renaming on model update

### DISCUSSION THREAD

**Discussion by kapishsingla** [eclipse-theia/theia#14674] [Q&A]
Hello Team,
Greetings of the day,

I am working on a Scenario in which fileName corresponds to `name` field in `Langium Model` . On rename of File using rename command, I am able to acheive the change in corresponding name field in Model and now I want to achieve the same result by change of name field and update the corresponding fileName(URI)

Just for better understanding, We can assume the java file in eclipse.. On change the className in java file, It correspondly changes the FileName. 

I have an approach which we can catch the `onUpdate` event on client for the change in document.

Is there any better approach to achieve this?

**Comment by msujew** [comment]
Hey @kapishsingla,

did you mean to post this to the [Langium discussion forum](https://github.com/eclipse-langium/langium/discussions)? I don't believe this is related to Theia.
